prometheus_xmpp/__main__.py

```diff
@@ -104,33 +119,64 @@ def read_password_from_command(cmd):
     Args:
         cmd: The command that should be executed.
     """
-    out = subprocess.check_output(cmd, shell=True).decode('utf-8')
-    lines = out.split('\n')
+    out = subprocess.check_output(cmd, shell=True).decode("utf-8")
+    lines = out.split("\n")
     first_line = lines[0]
 
     return first_line.strip()
 
 
 class XmppApp(slixmpp.ClientXMPP):
-
-    def __init__(self, jid, password_cb,
-                 amtool_allowed=None, alertmanager_url=None):
-
+    def __init__(
+        self,
+        jid,
+        password_cb,
+        amtool_allowed=None,
+        alertmanager_url=None,
+        muc=False,
+        muc_jid=None,
+        omemo=False,
+    ):
         password = password_cb()
 
         slixmpp.ClientXMPP.__init__(self, jid, password)
         self._amtool_allowed = amtool_allowed or []
         self.alertmanager_url = alertmanager_url
+        self.muc = muc
+        self.muc_jid = muc_jid
+        self.omemo = omemo
+        self.eme_ns = "eu.siacs.conversations.axolotl"  # use OMEMO
         self.auto_authorize = True
         self.add_event_handler("session_start", self.start)
         self.add_event_handler("message", self.message)
         self.add_event_handler("disconnected", self.lost)
         self.add_event_handler("failed_auth", self.failed_auth)
-        self.register_plugin('xep_0071')  # XHTML-IM
-        self.register_plugin('xep_0030')  # Service Discovery
-        self.register_plugin('xep_0004')  # Data Forms
-        self.register_plugin('xep_0060')  # PubSub
-        self.register_plugin('xep_0199')  # XMPP Ping
+        self.register_plugin("xep_0071")  # XHTML-IM
+        self.register_plugin("xep_0030")  # Service Discovery
+        self.register_plugin("xep_0004")  # Data Forms
+        self.register_plugin("xep_0060")  # PubSub
+        self.register_plugin("xep_0199")  # XMPP Ping
+        if self.muc:
+            self.register_plugin("xep_0045")  # Multi-User Chat
+        if self.omemo:
+            self.register_plugin("xep_0380")  # Explicit Message Encryption
+            OMEMO_DATA_DIR = os.path.join(
```

Comment:
This should probably be a configurable path; the path that the binary lives in won't be writeable in a lot of situations.